File: java/MASTER_SLAVE_WHOLE_PART_PROXY/SistemaAutorizacionPago.java
public class SistemaAutorizacionPago implements IAtorizacionPago{
    //Atributos
    Cargo cargoSolicitado = new Cargo();
    
    //Constructor
    public SistemaAutorizacionPago(){}

    //Métodos
    public void solicitaAutorizacionPago(Cargo cargo){
        System.out.println("\n\nSolicitando cargo");
        System.out.println("...");
        try{
            Thread.sleep(1000);
        }catch(InterruptedException e){

        }
        System.out.println("...");
        try{
            Thread.sleep(1000);
        }catch(InterruptedException e){
            
        }
        System.out.println("...");
        try{
            Thread.sleep(1000);
        }catch(InterruptedException e){
            
        }
        System.out.println("...");
        try{
            Thread.sleep(1000);
        }catch(InterruptedException e){
            
        }
        System.out.println("...");
        try{
            Thread.sleep(1000);
        }catch(InterruptedException e){
            
        }
        autorizacionCargo(cargo);
        
    }

    public boolean verificacionCargo(){
        if(cargoSolicitado.getcargoExitoso() == true){
            return true;
        }
        else{
            return false;
        }
    }

    public void autorizacionCargo(Cargo cargo){
        cargo.setCargoExitoso(true);
    }
}

File: php/MASTER_SLAVE_WHOLE_PART_PROXY/sistemaExterno/SistemaAutorizacionPago.php
<?php
require_once './pago/IAutorizacionPago.php';
class SistemaAutorizacionPago implements IAutorizacionPago{
    
    public  $cargoSolicitado;
    
    public function __construct() {
    }
    
    public  function solicitaAutorizacionPago(Cargo $cargo): bool {
        $this->cargoSolicitado = $cargo;
        
        echo '<br>Procesando cargo...';
        
        echo '<br>Procesando cargo...';
        $this->autorizacionCargo();
        $r = false;
        $r = $this->verificacionCargo();
        
        return $r;
        
    }
    
    public  function verificacionCargo(){
        $b = false;
        if ($this->cargoSolicitado->getCargoExitoso() == true){
            $b = true;
        }else
            $b = false;
        return $b;
    }
    
    public  function autorizacionCargo(){
        $this->cargoSolicitado->setCargoExitoso(true);
    }
    

}
